python-avd/pyavd/_anta/logs.py

```diff
@@ -33,36 +34,47 @@ class TestLoggerAdapter(LoggerAdapter):
     and prepend the message with the device and test names, and optionally the context: `<device> test context message`.
     """
 
-    def process(self, msg: LogMessage, kwargs: dict) -> tuple[str, dict]:
+    def process(self, msg: LogMessage | str, kwargs: MutableMapping[str, object]) -> tuple[str, MutableMapping[str, object]]:
         """Process the message and kwargs before logging."""
-        # Keep the extra dict in kwargs to pass it to the formatter if needed (following the standard LoggerAdapter behavior)
-        kwargs["extra"] = self.extra
+        # Ensure we always have a dict[str, object]
+        extra: dict[str, object] = dict(self.extra or {})
+        kwargs["extra"] = extra
 
-        # Extract the device, test, and context from extra
-        device = self.extra["device"]
-        test = self.extra["test"]
-        context = self.extra.get("context")
+        # Safely extract info
+        device = str(extra.get("device", "unknown"))
+        test = str(extra.get("test", "unknown"))
+        context = extra.get("context")
 
         prefix = f"<{device}> {test}"
-        if context:
+        if isinstance(context, str):
             prefix += f" {context}"
 
-        # Format the LogMessage using the provided kwargs and extract the fields name from the message string
-        fields = [field_name for _, field_name, _, _ in string.Formatter().parse(msg.value) if field_name is not None]
-        msg = msg.value.format(**kwargs)
+        # Support both LogMessage enums and plain strings
+        msg_value: str = msg.value if isinstance(msg, LogMessage) else str(msg)
 
-        # Removing the fields name from kwargs to preserve standard logging kwargs only that should always be passed through (e.g. exc_info, stack_info, etc.)
+        # Find format field names
+        fields = [field_name for _, field_name, _, _ in string.Formatter().parse(msg_value) if field_name is not None]
+
+        # Try formatting; if fails, keep raw text
+        try:
+            formatted_msg = msg_value.format(**kwargs)
+        except Exception:
+            formatted_msg = msg_value
+
+        # Remove used fields to not interfere with logging kwargs
         for field in fields:
             kwargs.pop(field, None)
 
-        return f"{prefix} {msg}", kwargs
+        return f"{prefix} {formatted_msg}", kwargs
 
     @contextmanager
     def context(self, context: str) -> Generator[TestLoggerAdapter, None, None]:
         """Temporarily add context to the logger."""
-        original_extra = dict(self.extra)
+        original_extra = dict(getattr(self, "extra", {}) or {})
```

Comment:
We need to keep the original_extra as None if it was None, not an empty dict.
```suggestion
        original_extra = dict(self.extra) if self.extra is not None else None
```

Reply:
updated